lib/Parser/Parse.cpp

```diff
@@ -14003,37 +13970,56 @@ DeferredFunctionStub * Parser::BuildDeferredStubTree(ParseNodeFnc *pnodeFnc, Rec
         }
 
         ParseNodeFnc* pnodeFncChild = pnodeChild->AsParseNodeFnc();
-        AnalysisAssertOrFailFast(i < nestedCount);
-
-        if (pnodeFncChild->pnodeBody != nullptr)
-        {
-            // Anomalous case of a non-deferred function nested within a deferred one.
-            // Work around by discarding the stub tree.
-            return nullptr;
-        }
+        AnalysisAssertOrFailFast(currentStubIndex < deferredStubCount);
+        Assert(pnodeFncChild->pnodeBody == nullptr);
 
         if (pnodeFncChild->IsGeneratedDefault())
         {
-            ++i;
+            ++currentStubIndex;
             pnodeChild = pnodeFncChild->pnodeNext;
             continue;
         }
 
-        deferredStubs[i].fncFlags = pnodeFncChild->fncFlags;
-        deferredStubs[i].nestedCount = pnodeFncChild->nestedCount;
-        deferredStubs[i].restorePoint = *pnodeFncChild->pRestorePoint;
-        deferredStubs[i].deferredStubs = BuildDeferredStubTree(pnodeFncChild, recycler);
-        deferredStubs[i].ichMin = pnodeChild->ichMin;
+        deferredStubs[currentStubIndex].fncFlags = pnodeFncChild->fncFlags;
+        deferredStubs[currentStubIndex].nestedCount = pnodeFncChild->nestedCount;
+        deferredStubs[currentStubIndex].restorePoint = *pnodeFncChild->pRestorePoint;
+        deferredStubs[currentStubIndex].deferredStubs = BuildDeferredStubTree(pnodeFncChild, recycler);
+        deferredStubs[currentStubIndex].ichMin = pnodeChild->ichMin;
 
         // Save the set of captured names onto the deferred stub.
         // Since this set is allocated in the Parser arena, we'll have to convert these
         // into indices in a string table which will survive when the parser goes away.
-        deferredStubs[i].capturedNamePointers = pnodeFncChild->GetCapturedNames();
+        deferredStubs[currentStubIndex].capturedNamePointers = pnodeFncChild->GetCapturedNames();
 
-        ++i;
+        ++currentStubIndex;
         pnodeChild = pnodeFncChild->pnodeNext;
     }
 
+    return currentStubIndex;
+}
+
+DeferredFunctionStub * Parser::BuildDeferredStubTree(ParseNodeFnc *pnodeFnc, Recycler *recycler)
+{
+    Assert(CONFIG_FLAG(ParserStateCache));
+
+    uint nestedCount = pnodeFnc->nestedCount;
+    if (nestedCount == 0)
+    {
+        return nullptr;
+    }
+
+    if (pnodeFnc->deferredStub)
+    {
+        return pnodeFnc->deferredStub;
+    }
+
+    DeferredFunctionStub* deferredStubs = RecyclerNewArray(recycler, DeferredFunctionStub, nestedCount);
+
+    uint currentStubIndex = BuildDeferredStubTreeHelper(pnodeFnc->pnodeScopes, deferredStubs, 0, nestedCount, recycler);
```

Comment:
>pnodeScopes [](start = 66, length = 11)

Do you have a test case with class definitions in param scope?

Reply:
Not one which gets serialized, I'll add one to this test.